applications/pelion_client/sample.yaml

```diff
@@ -0,0 +1,14 @@
+sample:
+  name: Pelion Client Application
+  description: Pelion conectivity reference application
+tests:
+  applications.pelion_client.zdebug:
+    build_only: true
+    build_on_all: true
```

Comment:
You can tell Twister (hence CI) not to treat warnings as errors by adding sth like this (you can be more specific with warnings to ignore):
`extra_args: EXTRA_CFLAGS=-Wall`